libs/libc/modlib/modlib_registry.c

```diff
@@ -82,40 +68,7 @@ static FAR struct module_s *g_mod_registry;
 
 void modlib_registry_lock(void)
 {
-  pid_t me;
-  int ret;
-
-  /* Do we already hold the semaphore? */
-
-  me = getpid();
-  if (me == g_modlock.holder)
-    {
-      /* Yes... just increment the count */
-
-      g_modlock.count++;
-      DEBUGASSERT(g_modlock.count > 0);
-    }
-
-  /* Take the semaphore (perhaps waiting) */
-
-  else
-    {
-      while ((ret = _SEM_WAIT(&g_modlock.lock)) < 0)
-        {
-          /* The only case that an error should occur here is if
-           * the wait was awakened by a signal.
-           */
-
-          DEBUGASSERT(_SEM_ERRNO(ret) == EINTR ||
-                      _SEM_ERRNO(ret) == ECANCELED);
```

Comment:
need to check `ECANCELED` path for mutex. I think it is safe, but need to verify

Reply:
you mean add `DEBUGASSERT` to `nxmutex` ？

Reply:
I mean that `modlib_registry_lock` spins in while until success and return code was checked vs `ECANCELED`. The `EINTR` is handled inside the `nxrmutex_lock`, so the question is only about `ECANCELED` and `DEBUGASSERT`.

Reply:
The same for `dns_semtake`

Reply:
I mean that it is not a question to you, but more. POSIX does not define `ECANCELED` error code for `sem_wait`, so I will start a thread in a mailing list to clarify why NuttX violate standard in this case

Reply:
ECANCELED/EINTR is better catch and ignore inside nxrmutex_lock.